examples/filesystem/main.c

```diff
@@ -0,0 +1,239 @@
+    return 0;
+#else
+    puts("No external flash file system selected");
+    return 1;
+#endif
+}
+
+static int _format(int argc, char **argv)
+{
+    (void)argc;
+    (void)argv;
+#if defined(MTD_0) && (defined(MODULE_SPIFFS) || defined(MODULE_LITTLEFS))
+    int res = vfs_format(&flash_mount);
+    if (res < 0) {
+        printf("Error while formatting %s\n", FLASH_MOUNT_POINT);
+        return 1;
+    }
+
+    printf("%s successfully formatted\n", FLASH_MOUNT_POINT);
+    return 0;
+#else
+    puts("No external flash file system selected");
+    return 1;
+#endif
+}
+
+static int _umount(int argc, char **argv)
+{
+    (void)argc;
+    (void)argv;
+#if defined(MTD_0) && (defined(MODULE_SPIFFS) || defined(MODULE_LITTLEFS))
+    int res = vfs_umount(&flash_mount);
+    if (res < 0) {
+        printf("Error while unmounting %s\n", FLASH_MOUNT_POINT);
+        return 1;
+    }
+
+    printf("%s successfully unmounted\n", FLASH_MOUNT_POINT);
+    return 0;
+#else
+    puts("No external flash file system selected");
+    return 1;
+#endif
+}
+
+static int _cat(int argc, char **argv)
+{
+    if (argc < 2) {
+        puts("filename missing");
+        return 1;
+    }
+    /* With newlib, low-level syscalls are plugged to RIOT vfs
+     * on native, open/read/write/close/... are plugged to RIOT vfs */
+#ifdef MODULE_NEWLIB
+    FILE *f = fopen(argv[1], "r");
+    if (f == NULL) {
+        printf("file %s does not exist\n", argv[1]);
+        return 1;
+    }
+    char c;
+    while (fread(&c, 1, 1, f) != 0) {
+        putchar(c);
+    }
+    fclose(f);
+#else
+    int fd = open(argv[1], O_RDONLY);
+    if (fd < 0) {
+        printf("file %s does not exist\n", argv[1]);
+        return 1;
+    }
+    char c;
+    while (read(fd, &c, 1) != 0) {
+        putchar(c);
+    }
+    close(fd);
+#endif
+    return 0;
+}
+
+static int _write(int argc, char **argv)
+{
+    if (argc != 3) {
+        printf("Usage: %s <file> <str>\n", argv[0]);
+        return 1;
+    }
+
+#ifdef MODULE_NEWLIB
+    FILE *f = fopen(argv[1], "w+");
+    if (f == NULL) {
+        printf("error while trying to create %s\n", argv[1]);
+        return 1;
+    }
+    if (fwrite(argv[2], 1, strlen(argv[2]), f) != strlen(argv[2])) {
+        puts("Error while writing");
+    }
+    fclose(f);
+#else
+    int fd = open(argv[1], O_RDWR | O_CREAT);
+    if (fd < 0) {
+        printf("error while trying to create %s\n", argv[1]);
+        return 1;
+    }
+    if (write(fd, argv[2], strlen(argv[2])) != (ssize_t)strlen(argv[2])) {
+        puts("Error while writing");
+    }
+    close(fd);
+#endif
+    return 0;
+}
+
+static const shell_command_t shell_commands[] = {
+    { "mount", "mount flash filesystem", _mount },
+    { "format", "format flash file system", _format },
+    { "umount", "unmount flash filesystem", _umount },
+    { "cat", "print the content of a file", _cat },
+    { "write", "write a string in a file", _write },
```

Comment:
I know it isn't like the original `tee` command (read from stdin and wride to file), but in conjunction with `cat` I really would like this command to be called `tee` :smile_cat:

Reply:
changed ;)